prereise/gather/demanddata/transportation_electrification/generate_scaling_factors.py

```diff
@@ -0,0 +1,210 @@
+    )
+
+    return (
+        df_ua.squeeze().loc[lambda x: x != "[no data]"],
+        df_state.rename(index=state2abv).squeeze(),
+    )
+
+
+def calculate_vmt_for_ua(census_ua, tht_ua):
+    """Calculate the total annual Vehicle Miles Traveled (VMT) in urban areas
+
+    :param dict census_ua: dictionary as returned by :func:`load_census_ua`
+    :param pandas.Series tht_ua: vmt per capita in urban areas as returned by
+        :func:`load_dot_vmt_per_capita`
+    :return: (*dict*) -- keys are state abbreviations and values are series giving
+        annual vmt by urban areas.
+    """
+
+    tht_ua_format = tht_ua.copy()
+    vmt_for_ua = {}
+
+    tht_ua_format.index = tht_ua_format.index.str.replace(r"[, -.]", "", regex=True)
+    format2original = dict(zip(tht_ua_format.index, tht_ua.index))
+
+    for s in census_ua:
+        census_ua_format = census_ua[s].copy()
+        census_ua_format.index = census_ua_format.index.str.replace(
+            r"[, -.]", "", regex=True
+        )
+        common = set(tht_ua_format.index).intersection(set(census_ua_format.index))
+        vmt_for_ua[s] = (
+            pd.DataFrame(
+                {
+                    "Annual VMT": [
+                        365 * tht_ua_format.loc[i] * census_ua_format.loc[i]
+                        for i in common
+                    ]
+                },
+                index=list(common),
+            )
+            .rename(index=format2original)
+            .squeeze()
+        )
+
+    return vmt_for_ua
+
+
+def calculate_vmt_for_state(census_state, tht_state):
+    """Calculate the total annual Vehicle Miles Traveled (VMT) in states
+
+    :param dict census_state: dictionary as returned by :func:`load_census_state`
+    :param pandas.Series tht_state: vmt per capita in states as returned by
+        :func:`load_dot_vmt_per_capita`
+    :return: (*pandas.Series*) -- indices are state abbreviations and values are annual
+        VMT in state.
+    """
+    common = list(set(tht_state.index).intersection(set(census_state.index)))
+    vmt_for_state = tht_state.loc[common] * census_state.loc[common]
+
+    return vmt_for_state
+
+
+def calculate_urban_rural_fraction(vmt_for_ua, vmt_for_state):
+    """Calculate the percentage of Vehicle Miles Traveled (VMT) in urban and rural areas
+
+    :param dict vmt_for_ua: dictionary as returned by :func:`calculate_vmt_for_ua`.
+    :param pandas.Series vmt_for_state: series as returned by
+        :func:`calculate_vmt_for_state`
+    :return: (*tuple*) -- keys are state abbreviations and values are either series of
+        percentage vmt in urban areas (first element) or percentage in rural area
+        (second element)
+    """
+    vmt_for_ua_perc = vmt_for_ua.copy()
+    vmt_for_ra_perc = {}
+    for s in vmt_for_ua:
+        if s in vmt_for_state.index:
+            vmt_for_ua_perc[s] = vmt_for_ua[s] / vmt_for_state.loc[s]
+            vmt_for_ra_perc[s] = 1 - vmt_for_ua_perc[s].sum()
+        else:
```

Comment:
maybe add a comment that this condition is for DC

Reply:
Done